packages/editor-ui/src/views/NodeView.vue

```diff
@@ -1275,344 +1199,397 @@ export default mixins(
+						if (this.dropPrevented) {
+							this.dropPrevented = false;
+							return;
+						}
 
-						// Change also the color to give an additional visual hint
-						info.connection.setPaintStyle({ strokeWidth: 2, stroke: '#334455' });
-					} else if (Math.abs(info.sourceEndpoint.anchor.lastReturnValue[1] - info.targetEndpoint.anchor.lastReturnValue[1]) < 30) {
-						info.connection.setConnector(['Straight']);
-					}
+						if (this.pullConnActiveNodeName) {
+							const sourceNodeName = this.$store.getters.getNodeNameByIndex(connection.sourceId.slice(NODE_NAME_PREFIX.length));
+							const outputIndex = connection.getParameters().index;
 
-					// @ts-ignore
-					info.connection.removeOverlay('drop-add-node');
-
-					if (this.isReadOnly === false) {
-						// Display the connection-delete button only on hover
-						let timer: NodeJS.Timeout | undefined;
-						info.connection.bind('mouseover', (connection: IConnection) => {
-							if (timer !== undefined) {
-								clearTimeout(timer);
-							}
-							const overlay = info.connection.getOverlay('remove-connection');
-							overlay.setVisible(true);
-						});
-						info.connection.bind('mouseout', (connection: IConnection) => {
-							timer = setTimeout(() => {
-								const overlay = info.connection.getOverlay('remove-connection');
-								overlay.setVisible(false);
-								timer = undefined;
-							}, 500);
+							this.connectTwoNodes(sourceNodeName, outputIndex, this.pullConnActiveNodeName, 0);
+							this.pullConnActiveNodeName = null;
+							return;
+						}
+
+						insertNodeAfterSelected({
+							sourceId: connection.sourceId,
+							index: connection.getParameters().index,
+							eventSource: 'node_connection_drop',
 						});
+					} catch (e) {
+						console.error(e);  // eslint-disable-line no-console
+					}
+				});
 
+				this.instance.bind('beforeDrop', (info) => {
+					try {
+						const sourceInfo = info.connection.endpoints[0].getParameters();
 						// @ts-ignore
-						info.connection.addOverlay([
-							'Label',
-							{
-								id: 'remove-connection',
-								label: '<span class="delete-connection clickable" title="Delete Connection">x</span>',
-								cssClass: 'remove-connection-label',
-								visible: false,
-								events: {
-									mousedown: () => {
-										this.__removeConnectionByConnectionInfo(info, true);
-									},
-								},
-							},
-						]);
+						const targetInfo = info.dropEndpoint.getParameters();
+
+						const sourceNodeName = this.$store.getters.getNodeNameByIndex(sourceInfo.nodeIndex);
+						const targetNodeName = this.$store.getters.getNodeNameByIndex(targetInfo.nodeIndex);
+
+						// check for duplicates
+						if (this.getConnection(sourceNodeName, sourceInfo.index, targetNodeName, targetInfo.index)) {
+							this.dropPrevented = true;
+							return false;
+						}
+
+						return true;
+					} catch (e) {
+						console.error(e);  // eslint-disable-line no-console
+						return true;
 					}
```

Comment:
why wrap all binds with try/catch? because jsplumb has try /catch at bind level but it removes details.. this makes it easier to pin down errors